server/src/main/java/org/elasticsearch/cluster/routing/allocation/allocator/BalancedShardsAllocator.java

```diff
@@ -745,10 +755,14 @@ public void moveShards() {
                     if (logger.isTraceEnabled()) {
                         logger.trace("Moved shard [{}] to node [{}]", shardRouting, targetNode.getRoutingNode());
                     }
+                    if (allocation.hasTargetRelocatingShards()) {
```

Comment:
There was a routingNodes.relocateShard call above with a target node. Why would we ever have to do this check? Can we just return true?

Reply:
Maybe this is related to the configurable number of relocations you have on the RoutingNodes -- I inquired about that on the `hasTargetRelocatingShards` method implementation.

Reply:
We decide to not making this value configurable as suggested. So it's no longer an issue here.